media-libs/libsdl2/patches/libsdl2-2.0.20.patchset

```diff
@@ -1265,3 +1265,145 @@ index c5026de..868512e 100644
  	mouse->FreeCursor = HAIKU_FreeCursor;
 -- 
 2.30.2
+
+
+From 369898e9f76829511e6cdd922866696d414aa963 Mon Sep 17 00:00:00 2001
+From: Ivan Holmes <ivan@ivanholmes.co.uk>
+Date: Sat, 7 May 2022 19:17:29 +0100
+Subject: Don't create a BApplication if one already exists
+
+
+diff --git a/src/main/haiku/SDL_BeApp.cc b/src/main/haiku/SDL_BeApp.cc
+index a5e1428..3b0f784 100644
+--- a/src/main/haiku/SDL_BeApp.cc
++++ b/src/main/haiku/SDL_BeApp.cc
+@@ -84,10 +84,12 @@ SDL_InitBeApp(void)
+ {
+     /* Create the BApplication that handles appserver interaction */
+     if (SDL_BeAppActive <= 0) {
++    	if (!be_app) {
```

Comment:
Combine this check with the previous one (i.e. `SDL_BeAppActive <= 0 && !be_app`)

Reply:
Ah yeah that would be cleaner, will do.

Reply:
Yeah, we shouldn't interfere with the working directory in that case.

Reply:
Makes sense, thanks